File: data/transcoder_evaluation_gfg/go/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(notes []int, n int) int {
	var (
		fiveCount int = 0
		tenCount  int = 0
	)
	for i := int(0); i < n; i++ {
		if notes[i] == 5 {
			fiveCount++
		} else if notes[i] == 10 {
			if fiveCount > 0 {
				fiveCount--
				tenCount++
			} else {
				return 0
			}
		} else {
			if fiveCount > 0 && tenCount > 0 {
				fiveCount--
				tenCount--
			} else if fiveCount >= 3 {
				fiveCount -= 3
			} else {
				return 0
			}
		}
	}
	return 1
}
//TOFILL
func main() {
	var (
		n_success int      = 0
		param0_0  []int   = []int{20}
		param0_1  []int   = []int{5, 5, 5, 20, 10}
		param0_2  []int  = []int{5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10}
		param0_3  []int  = []int{10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 18}
		param0_4  []int   = []int{5, 5, 20}
		param0_5  []int  = []int{10, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5}
		param0_6  []int  = []int{5, 10, 20, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5}
		param0_7  []int  = []int{-82, -10, -78, -84, 68, 62, 10, 20, -86, -98, 92, 70, 40, -12, -20, -36, 8, -70, 6, 8, 44, -24, 8, -18, 76, -54, -14, -94, -68, -62, -24, -36, -74, 92, 92, -80, 48, 56, 94}
		param0_8  []int  = []int{10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5}
		param0_9  []int  = []int{46, 46, 93, 57, 82, 34, 83, 80, 77, 36, 80, 85, 69, 28, 9, 56, 49, 27, 83, 25, 1, 80, 99, 14, 69, 82, 79, 71, 74, 34}
		param0    [][]int = [][]int{param0_0, param0_1, param0_2, param0_3, param0_4, param0_5, param0_6, param0_7, param0_8, param0_9}
		param1    []int  = []int{4, 5, 27, 12, 2, 17, 7, 31, 25, 20}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(([]int)(param0[i]), param1[i]) == f_gold(([]int)(param0[i]), param1[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut notes: *mut i32,
                                mut n: i32) -> i32 {
    let mut fiveCount: i32 = 0 as i32;
    let mut tenCount: i32 = 0 as i32;
    let mut i: i32 = 0 as i32;
    while i < n {
        if *notes.offset(i as isize) == 5 as i32 {
            fiveCount += 1
        } else if *notes.offset(i as isize) == 10 as i32 {
            if fiveCount > 0 as i32 {
                fiveCount -= 1;
                tenCount += 1
            } else { return 0 as i32 }
        } else if fiveCount > 0 as i32 && tenCount > 0 as i32
         {
            fiveCount -= 1;
            tenCount -= 1
        } else if fiveCount >= 3 as i32 {
            fiveCount -= 3 as i32
        } else { return 0 as i32 }
        i += 1
    }
    return 1 as i32;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0_0: [i32; 1] = [20 as i32];
    let mut param0_1: [i32; 5] =
        [5 as i32, 5 as i32, 5 as i32,
         20 as i32, 10 as i32];
    let mut param0_2: [i32; 33] =
        [5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32];
    let mut param0_3: [i32; 17] =
        [10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 18 as i32];
    let mut param0_4: [i32; 3] =
        [5 as i32, 5 as i32, 20 as i32];
    let mut param0_5: [i32; 22] =
        [10 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32];
    let mut param0_6: [i32; 15] =
        [5 as i32, 10 as i32, 20 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32];
    let mut param0_7: [i32; 39] =
        [-(82 as i32), -(10 as i32), -(78 as i32),
         -(84 as i32), 68 as i32, 62 as i32,
         10 as i32, 20 as i32, -(86 as i32),
         -(98 as i32), 92 as i32, 70 as i32,
         40 as i32, -(12 as i32), -(20 as i32),
         -(36 as i32), 8 as i32, -(70 as i32),
         6 as i32, 8 as i32, 44 as i32,
         -(24 as i32), 8 as i32, -(18 as i32),
         76 as i32, -(54 as i32), -(14 as i32),
         -(94 as i32), -(68 as i32), -(62 as i32),
         -(24 as i32), -(36 as i32), -(74 as i32),
         92 as i32, 92 as i32, -(80 as i32),
         48 as i32, 56 as i32, 94 as i32];
    let mut param0_8: [i32; 47] =
        [10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 10 as i32,
         10 as i32, 10 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32, 5 as i32,
         5 as i32, 5 as i32];
    let mut param0_9: [i32; 30] =
        [46 as i32, 46 as i32, 93 as i32,
         57 as i32, 82 as i32, 34 as i32,
         83 as i32, 80 as i32, 77 as i32,
         36 as i32, 80 as i32, 85 as i32,
         69 as i32, 28 as i32, 9 as i32,
         56 as i32, 49 as i32, 27 as i32,
         83 as i32, 25 as i32, 1 as i32,
         80 as i32, 99 as i32, 14 as i32,
         69 as i32, 82 as i32, 79 as i32,
         71 as i32, 74 as i32, 34 as i32];
    let mut param0: [*mut i32; 10] =
        [param0_0.as_mut_ptr(), param0_1.as_mut_ptr(), param0_2.as_mut_ptr(),
         param0_3.as_mut_ptr(), param0_4.as_mut_ptr(), param0_5.as_mut_ptr(),
         param0_6.as_mut_ptr(), param0_7.as_mut_ptr(), param0_8.as_mut_ptr(),
         param0_9.as_mut_ptr()];
    let mut param1: [i32; 10] =
        [4 as i32, 5 as i32, 27 as i32,
         12 as i32, 2 as i32, 17 as i32,
         7 as i32, 31 as i32, 25 as i32,
         20 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
